FinSearchUnified/Tests/Bundle/SearchBundleFindologic/QueryBuilderFactoryTest.php

```diff
@@ -0,0 +1,180 @@
+    {
+        parent::setUp();
+
+        $this->factory = new QueryBuilderFactory(
+            Shopware()->Container()->get('http_client'),
+            Shopware()->Container()->get('shopware_plugininstaller.plugin_manager'),
+            Shopware()->Container()->get('config')
+        );
+
+        /** @var ContextServiceInterface $contextService */
+        $contextService = Shopware()->Container()->get('shopware_storefront.context_service');
+        $this->context = $contextService->getShopContext();
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testCreateQueryWithoutConditions()
+    {
+        $criteria = new Criteria();
+
+        $query = $this->factory->createQuery($criteria, $this->context);
+        $params = $query->getParameters();
+
+        $this->assertArrayHasKey('group', $params, 'Usergroup was expected to be present in the parameters');
+        $this->assertArrayNotHasKey('attrib', $params, 'No attributes were expected to be present in the parameters');
+        $this->assertArrayNotHasKey('query', $params, 'No search query was expected to be present in the parameters ');
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testCreateQueryithConditions()
+    {
+        $criteria = new Criteria();
+        $criteria->addCondition(new CategoryCondition([5, 12]));
+        $criteria->addCondition(new PriceCondition(1, 20));
+        $criteria->addCondition(new ProductAttributeCondition('vendor', '=', 'Findologic Rockers'));
+        $criteria->addCondition(new SearchTermCondition('blubbergurke'));
+        $criteria->addCondition(new IsAvailableCondition());
+
+        $query = $this->factory->createQuery($criteria, $this->context);
+        $params = $query->getParameters();
+
+        $this->assertArrayHasKey('attrib', $params, 'Attributes were expected to be present in the parameters');
+        $attrib = $params['attrib'];
+
+        // CategoryCondition
+        $this->assertArrayHasKey('cat', $attrib, 'Category was expected to be present in the attribute parameters');
+        $this->assertEquals(
+            ['Genusswelten', 'Genusswelten_Tees und Zubehör_Tees'],
+            $attrib['cat'],
+            'Expected categories to contain the name of the provided category IDs'
+        );
+
+        // PriceCondition
+        $this->assertArrayHasKey('price', $attrib, 'Prices were expected to be present in the attribute parameters');
+        $this->assertEquals(1, $attrib['price']['min'], 'Expected minimum price to be 1');
+        $this->assertEquals(20, $attrib['price']['max'], 'Expected maximum price to be 20');
+
+        // ProductAttributeCondition
+        $this->assertArrayHasKey('vendor', $attrib, 'Expected "vendor" to be available in the attribute parameters');
+        $this->assertEquals(
+            ['Findologic Rockers'],
+            $attrib['vendor'],
+            'Expected vendor to be an array containing "Findologic Rockers"'
```

Comment:
It would say that this should either be changed to `"vendor"` or you change it to `vendor` in the previous assertion. Both options work for me but it should be consistent.